p2p/transport/quic/listener.go

```diff
@@ -19,18 +20,18 @@ var quicListen = quic.Listen // so we can mock it in tests
 
 // A listener listens for QUIC connections.
 type listener struct {
-	quicListener   quic.Listener
-	conn           pConn
-	transport      *transport
-	rcmgr          network.ResourceManager
-	privKey        ic.PrivKey
-	localPeer      peer.ID
-	localMultiaddr ma.Multiaddr
+	quicListener    quic.Listener
+	conn            pConn
+	transport       *transport
+	rcmgr           network.ResourceManager
+	privKey         ic.PrivKey
+	localPeer       peer.ID
+	localMultiaddrs map[quic.VersionNumber]ma.Multiaddr
```

Comment:
It's a bit cleaner when we access them: https://github.com/libp2p/go-libp2p/blob/marco/quic-v1/p2p/transport/quic/listener.go#L143

Otherwise we'd have to branch and update a variable. Not really a problem, just a bit less tidy imo. I don't have a strong opinion here so if you feel strongly we can use two vars.

Reply:
No strong opinion either, but using two vars would make the return order of `LocalMultiAddr` deterministic, which would be nice (see https://github.com/libp2p/go-libp2p/pull/1881#discussion_r1023319759).